syntax = "proto3";
package object_store;





/*
ListObjects请求
*/
message ListObjectsRequest {
    /*
    存储桶名称
    */
    string bucketName = 1 ;
    /*
    要列举的对象前缀
    */
    string objectPrefix = 2 ;
    /*
    true代表递归查找，false代表类似文件夹查找，以'/'分隔，不查子文件夹。
    */
    bool recursive = 3 ;
}

/*
ListObjects返回
*/
message ListObjectsResponse {
message Objects {
message Owner {
    /*
    用户名
    */
    string name = 1 ;
    /*
    用户ID
    */
    string id = 2 ;
}
        
    
    /*
    对象MD5值
    */
    string etag = 1 ;
    /*
    对象名称
    */
    string name = 2 ;
    /*
    最后修改时间
    */
    string lastModified = 3 ;
    /*
    对象大小
    */
    int32 size = 4 ;
    /*
    标准MIME类型，描述对象数据格式
    */
    string contentType = 5 ;
    /*
    过期时间(超过将不再缓存)
    */
    string expires = 6 ;
    /*
    所属用户
    */
    Owner owner = 7 ;
}
        
    
    /*
    对象数组
    */
    repeated Objects objects = 1 ;
}

/*
ListObjectsApi返回
*/
message ListObjectsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    object_store.ListObjectsResponse data = 4 ;
}

